sauce/features/general/import-notification/index.js

```diff
@@ -0,0 +1,72 @@
+import { Feature } from 'core/feature';
+// import * as toolkitHelper from 'helpers/toolkit';
+
+export class ImportNotification extends Feature {
+  injectCSS() { return require('./index.css'); }
+
+  constructor() {
+    super();
+    this.isActive = false;
+    this.importClass = 'import-notification';
+    this.invoke = this.invoke.bind(this);
+  }
+
+  willInvoke() {
+    if (this.settings.enabled !== '0') {
+      if (this.settings.enabled === '2') {
+        this.importClass += '-red';
+      }
+
+      // Hook transaction imports so that we can run our stuff when things change
+      ynab.YNABSharedLib.defaultInstance.entityManager._transactionEntityPropertyChanged.addHandler(this.invoke);
+    }
+  }
+
+  shouldInvoke() {
+    return this.settings.enabled !== '0';
+  }
+
+  invoke() {
+    if (!this.isActive) {
```

Comment:
It's there because the .`remove()` and `.append()` functions result in observe() being fired which results in a loop. With the guard in place `checkImportsTransactions` won't be called while the classes are being "processed".  That logic was required before refactoring so if things are different in the new framework it may not be needed.